### ISSUE TITLE
Add other ALU ops to ASM

### ISSUE THREAD

**Issue by freesig** [essential-contributions/essential-base#165]

**Comment by freesig** [comment]
@mohammadfawaz I think this is all the missing arithmetic and logic ops. Can you think of any others?

**Comment by mohammadfawaz** [comment]
This is good! Let's start with this and we'll add more as we see more applications. I suspect logarithms will come in handy at some point.

**Comment by freesig** [comment]
Note to self, make room at end of arithmetic byte for more ops

**Comment by freesig** [comment]
@mohammadfawaz I was thinking maybe we want a bitwise AND for words?
The left and right shift only seem useful if you can combine two words into one after shifting things around.

**Comment by mohammadfawaz** [comment]
> @mohammadfawaz I was thinking maybe we want a bitwise AND for words?

Yeah those are standard I suppose.. wouldn't hurt to have them. They could be useful for various optimizations as well.
